wrapcheck/wrapcheck.go

```diff
@@ -91,7 +93,24 @@ func NewAnalyzer(cfg WrapcheckConfig) *analysis.Analyzer {
 }
 
 func run(cfg WrapcheckConfig) func(*analysis.Pass) (interface{}, error) {
+	var Err error
+
+	for i := range cfg.IgnoreSigRegexps {
```

Comment:
Is there any functional difference between doing this here (within run()) versus doing this within the anonymous function below? If not, why not just do it down there and eliminate the error variable above.

Reply:
Good comment! This is perfect location for initial parsing because you used closure to pass cfg down the scope, it is possible to run the initial compilation.  If we use `run` to compile regexes, we have two choices:
1) to run compilation on each `run` ( function `run` called on each file linter checking) (which is better than run compilation on each error check but still considered `bad`)
2) wrap it with something like `sync.Once`, but it's will look in general in the same way.